src/core/src/layout.cpp

```diff
@@ -638,6 +638,7 @@ void AttributeAdapter<ov::Layout>::set(const std::string& value) {
 }
 
 bool LayoutAttribute::visit_attributes(AttributeVisitor& visitor) {
+    std::lock_guard<std::mutex> lock{*m_mutex.get()};
```

Comment:
@songbell could you explain how it should solve the issue?

Reply:
I analyzed from the limited trace from user, it should be the race condition issue happened in ov::Hash pass, because auto will do thread compilation to multiple devices. I have enabled tests, please help a check.